src/Model.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Intermax\EloquentJsonApiClient;
+
+use Carbon\Carbon;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use Intermax\EloquentJsonApiClient\Attributes\DateTime;
+use Intermax\EloquentJsonApiClient\Attributes\Property;
+use Intermax\EloquentJsonApiClient\Attributes\Relationship;
+use ReflectionClass;
+use ReflectionProperty;
+
+abstract class Model
+{
+    protected int $perPage = 30;
+
+    final public function __construct()
+    {
+    }
+
+    public function uri(): string
+    {
+        return Str::of((new ReflectionClass($this))->getShortName())
+            ->lower()
+            ->plural()
+            ->prepend('/')
+            ->toString();
+    }
+
+    public function perPage(): int
+    {
+        return $this->perPage;
```

Comment:
🐮 we could just do it like this and remove one more property.

```suggestion
        return 30;
```

Reply:
You can extend with the method or the property now. Only the method is fine, but I'm not sure if it makes a real difference. For now I'm merging it so we can use it faster.